p2psrv/options.go

```diff
@@ -40,11 +40,6 @@ type Options struct {
 
 	KnownNodes Nodes
 
-	// BootstrapNodes are used to establish connectivity
-	// with the rest of the network using the V5 discovery
-	// protocol.
-	BootstrapNodes Nodes
-
```

Comment:
This comment is outdated BootstrapNodes are only ever used to pass nodes from the previous `p2pcomm` to the p2psrv, at which point they would be appended with the `KnownNodes` into the `s.bootstrapNodes`

Reply:
Before:
- BootstrapNodes + KnownNodes were used to discover new nodes
- KnownNodes were used to connect and discover new nodes + p2p communication

Now:
- KnowNodes are used to discover new nodes  + p2p communication

This cleans up the process. The idea behind this is that it should be the serving node responsibility to serve JUST discovery or discovery + p2p. It should not be the client's responsibility, because this line of action does not protect discovery nodes.

Reply:
I might be against this idea, `BootstrapNodes` and `KnownNodes` serves their different purpose, shouldn't mix them together. 

`BootStrapNodes` is for discovery protocol.
`KnownNodes` is known(connected or node operator supplied) peer-to-peer nodes.

Reply:
Yeah I understand that they serve different purposes, I just think that the idea behind it should be cleaner or simpler.

Why can't the `Bootstrap nodes` also be connected over p2p ? 
I guess in my head, it should be the receiving node responsibility to accept or deny the protocols it talks in.

Reply:
This was bugging me, took me a bit to figure out why, thanks for the patience and the review.

Fundamentally, I was adding a new feature in the middle of a clean-up/refactor - which I believe can arguably be a good simplification - but not the point of the PR.
The point of the PR was to make the flow of the usage of flags easy to follow and to have a testable p2p package.
I've :
- Reverted the removal of `BootstrapNodes` (no new feature)
- Renamed `BootstrapNodes` to DiscoveryNodes; `RemoteBootstrap` to `RemoteDiscoveryList` and `fallbackBootstrapNodes` to `fallbackDiscoveryNodes`
- Updated the tests

The functionality remains the same, but this is fine-tuning the names. It feels to me that DiscoveryNodes is a less prone to confusion name for the initial default connection nodes exclusively used for peer discovery (and never used for p2p).